### ISSUE TITLE
Show Tag's Color in Admin Page

### ISSUE THREAD

**Issue by maxfarrior** [jonaswinkler/paperless-ng#448]
I would like to propose that the color assigned to a tag be displayed on the tag editing page.

I am looking at the http://\<server\>/admin/documents/tag/ page. In the "Color" column, I see the following
![Paperless NG Tag Color Column](https://user-images.githubusercontent.com/26314838/105874103-38729d00-5fca-11eb-9296-7775e92514b1.png)

The issue is that I have no idea what those colors look like.

Yes, I could look up the colors, but that is extremely cumbersome. I'd have to look up every color on the list and make a decision by navigating through the x number of tabs that I've just opened.

The [original Paperless](https://github.com/the-paperless-project/paperless) showed the color in a small box next to the hex code. 
![Original Paperless Tags Column](https://user-images.githubusercontent.com/26314838/105874781-ff86f800-5fca-11eb-9556-9b4cb9e48c26.png)

That makes choosing the color very quick and easy. I would like to propose that this be added to Paperless-ng.

**Related Proposal 1**

This is closely related to the first proposal. It would be very nice to have a visual representation of each color *in the dropdown menu*.

In the Original Paperless, you could see each color by simply selecting it in the dropdown. You could scroll through each item to see all the options.

It would be *even easier* to pick a color if all of the colors are shown by just opening the dropdown menu. This avoids having to select each color to see what the choices are.

I cobbled something together to illustrate the idea.
![paperless-tag-menu-with-colors](https://user-images.githubusercontent.com/26314838/105878239-d0728580-5fce-11eb-91e2-8c307946937e.png)

**Related Proposal 2**

This is another proposal that's related to the first two. To make the color selection process easy, some websites have a small popup with a color wheel in it that can be used to graphically choose a color.

Adding such a feature would negate the need for a list of pre-defined colors. Just put a text field in the "color" tab, and when clicked, the pop up color wheel can be used to select a color, and the selected color's hex code will be inserted into the box.

I can't find an example right now, but if I see one, I'll edit. I think most people have at least seen this on other sites.

Happy to add more info if needed.

**Comment by shamoon** [comment]
Colors are available in the -ng tags administration area, I think the goal is to eventually hide this Django admin area completely. And with respect to that there are some ideas / plans for changing color assignment, selection etc

**Comment by maxfarrior** [comment]
>Colors are available in the -ng tags administration area

Forgive me, but you tell me where this is?

I'm currently looking at http://\<server>\/admin/documents/tag/, which I thought was the administration page, and I'm not seeing any colors shown.

**Comment by shamoon** [comment]
If you click on "Tags" on the left sidebar, e.g. /tags
<img width="1752" alt="Screen Shot 2021-01-26 at 9 47 33 AM" src="https://user-images.githubusercontent.com/4887959/105883274-859b4280-5fbb-11eb-8925-629f844bbd13.png">

**Comment by jonaswinkler** [comment]
Hey!

> The [original Paperless](https://github.com/the-paperless-project/paperless) showed the color in a small box next to the hex code.

I deliberately removed pretty much all custom html from the admin, for a couple reasons:
* Some were causing trouble with Django 3.1.
* The front end should be the primary interface. Some functionality isn't yet available there (mail).


> This is another proposal that's related to the first two. To make the color selection process easy, some websites have a small popup with a color wheel in it that can be used to graphically choose a color.

That's what I'll eventually do. It's just that this does not have a pretty high priority right now.

**Comment by maxfarrior** [comment]
@shamoon

Ahh, I see now. There are two places that the color of tags can be edited. The first is what you showed me, at http://\<server\>/tags and the second is what I stumbled on at http://\<server\>/admin/documents/tag/.

If http://\<server\>/admin/documents/tag/ is the administration page for Django, then I agree that hiding that would be best. It's not as polished, and new users could get confused (like me) when there are two places to make the same change.

@jonaswinkler 

I now understand that the /admin section is a part of Django, not the main application. That wasn't clear to me at first, and I now see that the actual application shows the colors of the tags, and that's what I should have been using to start with.

Also, good to hear that the color wheel idea was already on your mind.

I'll close this out, I see where I went wrong. Thank you for the help.
